src/reactapp/src/components/paypalExpress/hooks/usePaypalExpress.js

```diff
@@ -0,0 +1,149 @@
+import { useCallback, useEffect, useState } from 'react';
+import _get from 'lodash.get';
+import createCustomerToken from '../../../api/paypalExpress/createCustomerToken';
+import { __ } from '../../../i18n';
+import setPaymentMethodPaypalExpress from '../../../api/paypalExpress/setPaymentMethod';
+import placeOrder from '../../../api/cart/placeOrder';
+import LocalStorage from '../../../utils/localStorage';
+import { config } from '../../../config';
+import usePaypalExpressCartContext from './usePaypalExpressCartContext';
+
+export default function usePaypalExpress({ paymentMethod }) {
+  const [processPaymentEnable, setProcessPaymentEnable] = useState(false);
+  const {
+    hasCartBillingAddress,
+    selectedShippingMethod,
+    selectedPaymentMethod,
+    setPageLoader,
+    setErrorMessage,
+    cartId,
+  } = usePaypalExpressCartContext();
+  const query = window.location.search;
+  const selectedShippingMethodCode = _get(selectedShippingMethod, 'methodCode');
+  const selectedPaymentMethodCode = _get(selectedPaymentMethod, 'code');
+  const { code } = paymentMethod;
```

Comment:
Again `code` is not descriptive. Also, we don't want to pass the `payment method entirely. Seems this hook only uses the code. So pass only the code to the hook. maybe we can use a variable `paypalExpressCode` which clearly explains what it is.